.github/workflows/abi-diff.yml

```diff
@@ -0,0 +1,159 @@
+          conan profile update settings.compiler.libcxx=libstdc++11 default
+          conan remote add -f dice-group https://conan.dice-research.org/artifactory/api/conan/tentris
+          
+      - name: Cache conan data
+        id: cache-conan
+        uses: actions/cache@v3
+        with:
+          path: ~/.conan/data
+          key: conan-${{ inputs.os }}-${{ inputs.compiler }}-abi-diff
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0  # needed to check out all branches for this Action to work
+
+      - name: Run diff
+        id: diff
+        run: |
+          git diff remotes/origin/${{ inputs.base-branch }} -- ${{ inputs.search-path }} > /tmp/diff.txt
+          diff_lines=$(cat /tmp/diff.txt | wc -l)
+          echo "Diff has $diff_lines lines"
+          echo "nlines=$diff_lines" >> "$GITHUB_OUTPUT"
+
+      # Run cmake because the version file might be generated by it
+      - name: Configure CMake
+        env:
+          CC: ${{ steps.install_cc.outputs.cc }}
+          CXX: ${{ steps.install_cc.outputs.cxx }}
+        run: |
+          cmake -G Ninja -B build_dir
+
+      - name: Create ABI version detection program
+        run: |
+          PROG=$(cat << EOF
+          #include <iostream>
+          #include <${{ inputs.abi-version-header }}>
+          
+          int main() {
+            std::cout << ${{ inputs.abi-version-const }} << std::endl;
+          }
+          EOF
+          )
+          
+          echo "$PROG" > /tmp/detect-abi-version.cpp
+
+      - name: Fetch current ABI version
+        id: current-abi-version
+        run: |
+          ${{ steps.install_cc.outputs.cxx }} -I. -o /tmp/detect-abi-version-current /tmp/detect-abi-version.cpp
+          abi_version=$(/tmp/detect-abi-version-current)
+          
+          echo "Current ABI version: $abi_version"
+          echo "abi_version=$abi_version" >> "$GITHUB_OUTPUT"
+
+      - name: Fetch base branch ABI version
+        id: base-abi-version
+        run: |
+          git switch ${{ inputs.base-branch }}
+          ${{ steps.install_cc.outputs.cxx }} -I. -o /tmp/detect-abi-version-base /tmp/detect-abi-version.cpp
+          abi_version=$(/tmp/detect-abi-version-base)
+          
+          echo "Base ABI version: $abi_version"
+          echo "abi_version=$abi_version" >> "$GITHUB_OUTPUT"
+
+      - name: Check missing ABI version bump
+        if: ${{ steps.diff.outputs.nlines != 0 && steps.current-abi-version.outputs.abi_version == steps.base-abi-version.outputs.abi_version }}
+        run: |
+          echo "::error::Files changed but ABI version was not changed"
+          exit 1
+
+      - name: Check erroneous ABI version bump
+        if: ${{ steps.diff.outputs.nlines == 0 && steps.current-abi-version.outputs.abi_version != steps.base-abi-version.outputs.abi_version }}
+        run: |
+          echo "::error::Files did not change but ABI version was changed"
+          exit 1
```

Comment:
I am pondering if there can be a good reason why one would increase the ABI version even though non of the tracked files changes. 

The one thing I have in mind are changes to rdf4cpp that change binary layouts in tentris. We still have the option to dismiss this actions output. So, let's keep it as it is